### ISSUE TITLE
KDE Plasma - Emulator fails to launch in a Wayland session

### ISSUE THREAD

**Issue by aquova** [punesemu/puNES#232]
I have been a happy user of puNES for some time, but recently when I started using Wayland on my desktop machine, I noticed that the emulator fails to launch. Running it via a terminal session yields no logs at all, the program halts silently almost immediately. If I switch back to an X11 session, the emulator runs as expected. Running `echo $?` gives an exit code of 0, but other than that I'm not sure how else to get any debugging information.

Specs:
OS: Arch Linux
DE: KDE Plasma Wayland
Kernel: 5.19.12
CPU: Ryzen 5 5600X
GPU: Radeon RX 6650 XT

**Comment by punesemu** [comment]
But have you tried version 0.109 or WIP?

**Comment by punesemu** [comment]
ok, I'll try on a virtual machine and let you know.

**Comment by punesemu** [comment]
I was very fast. I installed kde plasma wayland on my virtual archlinux, compiled and started the emulator correctly. The only thing I haven't done is use the https://aur.archlinux.org/packages/punes-git but I used the sources directly.

**Comment by punesemu** [comment]
The package xorg-wayland is installed?

**Comment by punesemu** [comment]
I've tested to compile via https://aur.archlinux.org/packages/punes-git and again it worked.

**Comment by aquova** [comment]
Output here: [ldd.txt](https://github.com/punesemu/puNES/files/9708936/ldd.txt)

At a quick glance, everything seems to look fine here. After work, I'll try and compile it myself and see if I can't determine why it's exiting early.

**Comment by punesemu** [comment]
>Output here: [ldd.txt](https://github.com/punesemu/puNES/files/9708936/ldd.txt)

I don't see problem.
>At a quick glance, everything seems to look fine here. After work, I'll try and compile it myself and see if I can't determine why it's exiting early.

Let me know if you find anything. Unfortunately, without being able to reproduce the problem, I don't know how to help you. Thanks anyway for the time you are dedicating.

**Comment by aquova** [comment]
Well, I've discovered what the issue was, and the emulator is now working perfectly fine, although I'm a little confused as to how I got here. The emulator was exiting in `core/main.c:86`, due to `gui_control_instance()` returning `EXIT_ERROR`. When I went to check that function, `qt.app->isSecondary()` was returning true. Odd. I hadn't rebooted today, so I ran `ps aux | grep punes` and sure enough there several zombie `punes` processes running. `killall punes` cleared them all out, and after that both the package from the AUR and the one I compiled myself are running perfectly fine.

I guess my older version of puNES that I had this morning never exited correctly, and so when I reinstalled the `punes-git` package, it just exited immediately due to another instance "existing" already. If I had rebooted, that would've solved the issue as well.

I'm going to close this then, as it's not really a bug, and I guess would only affect people going from older to newer versions without a reboot in between. Hopefully this is useful if anyone else finds themselves here, and thank you for your time.
